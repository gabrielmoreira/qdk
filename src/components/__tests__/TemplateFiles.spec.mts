import { resetFilesystem, toSnapshot } from '#test/helpers.mjs';
import { describe, expect, it, vi } from 'vitest';
import { Project, TemplateFiles } from '../../index.mjs';

vi.mock('node:fs');
vi.mock('node:fs/promises');

describe('components/TemplateFiles', () => {
  it('', async () => {
    // Given
    // ... filesystem has some templates
    resetFilesystem({
      json: {
        '/tmp/template-files/templates/hello.js.eta':
          'export const hello = "<%= it.name %>"',
        '/tmp/template-files/templates/some/directory/world.js.eta':
          'export const world = "<%= it.name %>"',
      },
    });
    // ... a new project located at '/tmp/template-files/project'
    const testProject = Project.create({
      name: 'template-files',
      cwd: '/tmp/template-files/project',
    });
    // ... and templates configured at '/tmp/template-files/templates/'
    new TemplateFiles(
      testProject,
      {
        templatePath: '/tmp/template-files/templates/',
      },
      // and some dafault data
      { name: 'Gabriel' },
    );

    // When
    await testProject.synth();

    // Then
    expect(toSnapshot()).toMatchSnapshot();
  });
});
