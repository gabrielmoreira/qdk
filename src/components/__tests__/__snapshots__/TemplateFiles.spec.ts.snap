// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/TemplateFiles >  1`] = `
{
  "/tmp/template-files/project/.qdk/meta.json": "{
  "project": "template-files",
  "custom": {},
  "files": [
    "./.qdk/meta.json",
    "./.qdk/tree.txt",
    "./hello.js",
    "./some/directory/world.js"
  ],
  "subprojects": []
}
",
  "/tmp/template-files/project/.qdk/tree.txt": "·
└── Project(template-files)
    ├── JsonFile(.qdk/meta.json)
    └── TemplateFiles
        ├── TextFile(hello.js)
        └── TextFile(some/directory/world.js)",
  "/tmp/template-files/project/hello.js": "export const hello = 'Gabriel';
",
  "/tmp/template-files/project/some/directory/world.js": "export const world = 'Gabriel';
",
  "/tmp/template-files/templates/hello.js.eta": "export const hello = "<%= it.name %>"",
  "/tmp/template-files/templates/some/directory/world.js.eta": "export const world = "<%= it.name %>"",
}
`;
